### DISCUSSION TITLE
[Docs] Enhance documentation on styling or using frontend frameworks

### DISCUSSION THREAD

**Discussion by laundmo** [lona-web-org/lona#97] [Feature Request]
Currently, it's not clear how I would use a frontend framework like Bootstrap with lona.

Could the documentation on this be improved, to include a full example of how lona would be used with something like bootstrap, tailwind(if possible, likely needs custom integration due to the css being compiled), bulma etc?

**Comment by fscherf** [comment]
Hi @laundmo,

I currently work on a lona-bootstrap5 package (https://github.com/lona-web-org/lona-bootstrap-5). This will be the example integration of an full-fledged frontend framework.

Chart.js is already integrated (https://github.com/lona-web-org/lona-chartjs)

**Comment by fscherf** [comment]
@laundmo: I updated [lona_bootstrap_5](https://github.com/lona-web-org/lona-bootstrap-5/tree/fscherf/dev%2Fmaster) and will release after the next Lona release, because the new widgets depend on some new Lona API (#34 ).

Do you think an example integration like this plus https://lona-web.org/end-user-documentation/frontends.html and https://lona-web.org/end-user-documentation/html.html#frontend-widgets are enough information?

**Comment by laundmo** [comment]
not really.

the issue is there the docs show all the concepts individually but never how they can work together coherently. adding a page explaining the basics of creating something like lona bootstrap would really tie things together.

our maybe just improving the docs pages you linked.

i think the underlying issue is that you obviously know the big picture of how it fits together, but its less obvious to anyone else reading the docs.

**Comment by fscherf** [comment]
I see. I will add an article to the cookbook section named "Integrating a third party library", explaining the steps i have taken to create ``lona_bootstrap_5``

**Comment by steezeburger** [comment]
I agree that a page showing all the setup code together would be really helpful. The documentation for custom frontends is kind of far down, and I feel some users won't take the time to make it there. I would be happy to help here! I plan on using this for a personal project.

**Comment by fscherf** [comment]
@steezeburger Some help would be great! :) Sorry that it takes me so long, there was much bug fixing required the last days. This is the next thing i will work on

**Comment by fscherf** [comment]
@laundmo, @steezeburger: I started a cookbook article. Could you give some feedback if this is the right direction?
